File: node/tfm.js
#!/usr/bin/env node

const fs = require("fs");
const https = require("https");
const { execSync } = require("child_process");

// -----------------
// Helper Functions
// -----------------

function listFiles() {
  fs.readdirSync(".").forEach(f => console.log(f));
}

function download(url, name) {
  if (!name) name = url.split("/").pop();
  const file = fs.createWriteStream(name);
  https.get(url, res => {
    res.pipe(file);
    file.on("finish", () => console.log("Downloaded:", name));
  });
}

function zipFolder(folder) {
  try {
    execSync(`zip -r "${folder}.zip" "${folder}"`);
    console.log("Zipped:", folder + ".zip");
  } catch (err) {
    console.error("Error zipping folder:", err.message);
  }
}

function unzipFile(file) {
  try {
    execSync(`unzip "${file}"`);
    console.log("Unzipped:", file);
  } catch (err) {
    console.error("Error unzipping file:", err.message);
  }
}

function deleteFileOrFolder(target) {
  try {
    fs.rmSync(target, { recursive: true, force: true });
    console.log("Deleted:", target);
  } catch (err) {
    console.error("Error deleting:", err.message);
  }
}

// -----------------
// Main
// -----------------

const args = process.argv.slice(2);
if (args.length < 1) {
  console.log("Usage: tfm <command> [args]");
  process.exit(1);
}

const cmd = args[0];

switch (cmd) {
  case "list":
    listFiles();
    break;
  case "download":
    if (!args[1]) {
      console.log("Usage: tfm download <url>");
      process.exit(1);
    }
    download(args[1]);
    break;
  case "zip":
    if (!args[1]) {
      console.log("Usage: tfm zip <folder>");
      process.exit(1);
    }
    zipFolder(args[1]);
    break;
  case "unzip":
    if (!args[1]) {
      console.log("Usage: tfm unzip <file.zip>");
      process.exit(1);
    }
    unzipFile(args[1]);
    break;
  case "delete":
    if (!args[1]) {
      console.log("Usage: tfm delete <file/folder>");
      process.exit(1);
    }
    deleteFileOrFolder(args[1]);
    break;
  default:
    console.log("Unknown command:", cmd);
    console.log("Available commands: list, download, zip, unzip, delete");
}

File: go/TFM.go
package main

import (
	"fmt"
	"io"
	"net/http"
	"os"
	"os/exec"
	"strings"
)

func listFiles() {
	files, _ := os.ReadDir(".")
	for _, f := range files {
		fmt.Println(f.Name())
	}
}

func download(url string) {
	name := url[strings.LastIndex(url, "/")+1:]
	resp, err := http.Get(url)
	if err != nil {
		fmt.Println("Download error:", err)
		return
	}
	defer resp.Body.Close()

	file, err := os.Create(name)
	if err != nil {
		fmt.Println("File create error:", err)
		return
	}
	defer file.Close()

	io.Copy(file, resp.Body)
	fmt.Println("Downloaded:", name)
}

func zipFolder(folder string) {
	err := exec.Command("zip", "-r", folder+".zip", folder).Run()
	if err != nil {
		fmt.Println("Zip error:", err)
		return
	}
	fmt.Println("Zipped:", folder+".zip")
}

func unzipFile(file string) {
	err := exec.Command("unzip", file).Run()
	if err != nil {
		fmt.Println("Unzip error:", err)
		return
	}
	fmt.Println("Unzipped:", file)
}

func deleteFileOrFolder(target string) {
	err := os.RemoveAll(target)
	if err != nil {
		fmt.Println("Delete error:", err)
		return
	}
	fmt.Println("Deleted:", target)
}

func main() {
	if len(os.Args) < 2 {
		fmt.Println("Usage: tfm <command> [args]")
		return
	}

	cmd := os.Args[1]

	switch cmd {
	case "list":
		listFiles()
	case "download":
		if len(os.Args) < 3 {
			fmt.Println("Usage: tfm download <url>")
			return
		}
		download(os.Args[2])
	case "zip":
		if len(os.Args) < 3 {
			fmt.Println("Usage: tfm zip <folder>")
			return
		}
		zipFolder(os.Args[2])
	case "unzip":
		if len(os.Args) < 3 {
			fmt.Println("Usage: tfm unzip <file.zip>")
			return
		}
		unzipFile(os.Args[2])
	case "delete":
		if len(os.Args) < 3 {
			fmt.Println("Usage: tfm delete <file/folder>")
			return
		}
		deleteFileOrFolder(os.Args[2])
	default:
		fmt.Println("Unknown command:", cmd)
		fmt.Println("Available commands: list, download, zip, unzip, delete")
	}
}
